utilities/transactions/transaction_base.cc

```diff
@@ -215,14 +217,21 @@ Status TransactionBaseImpl::Get(const ReadOptions& read_options,
 Status TransactionBaseImpl::GetForUpdate(const ReadOptions& read_options,
                                          ColumnFamilyHandle* column_family,
                                          const Slice& key, std::string* value,
-                                         bool exclusive) {
-  Status s = TryLock(column_family, key, true /* read_only */, exclusive);
+                                         bool exclusive, bool skip_validate) {
+  Status s = TryLock(column_family, key, true /* read_only */, exclusive,
+                     skip_validate);
 
   if (s.ok() && value != nullptr) {
     assert(value != nullptr);
     PinnableSlice pinnable_val(value);
     assert(!pinnable_val.IsPinned());
-    s = Get(read_options, column_family, key, &pinnable_val);
+    if (skip_validate && read_options.snapshot != nullptr) {
+      ReadOptions ro(read_options);
+      ro.snapshot = nullptr;
```

Comment:
Can you explain why we need to do it?

Reply:
If the caller expects to read the latest, not from the snapshot, should they not set `read_options.snapshot`? Users passing in a snapshot but we have an argument to skip it is too complicated to me.

Reply:
@hermanlee @lth would it work if we return error if snapshot and skip_validate both are set?

Reply:
In that case we do not need to extend ::GetForUpdate signature with a skip_validate boolean; setting the snapshot to nullptr will already have the same effect.